plugins/rdsr-ps/pffb/pff-ps-pffb.c

```diff
@@ -0,0 +1,1169 @@
+
+/* Attribute used in PFFB policy. */
+struct pffb_attribute {
+	struct attribute attr;
+	ssize_t (* show)(
+		struct kobject * kobj,
+		struct attribute * attr,
+		char * buf);
+	ssize_t (* store)(
+		struct kobject * foo,
+		struct attribute * attr,
+		const char * buf,
+		size_t count);
+};
+
+/* Show procedure! */
+static ssize_t pffb_attr_show(
+	struct kobject * kobj,
+	struct attribute * attr,
+	char * buf) {
+
+	if(strcmp(attr->name, "fwd_type") == 0) {
+		return snprintf(buf, PAGE_SIZE, "%u\n", pffb_distr);
+	}
+
+	return 0;
+}
+
+/* Store procedure! */
+static ssize_t pffb_attr_store(
+	struct kobject * kobj,
+	struct attribute * attr,
+	const char * buf,
+	size_t count) {
+
+	int ft = 0;
+	int op = 0;
+
+        /* Cleanup must happens before using a different strategy!
+        pffb_cleanup = 1;
+         */
+
+	if(strcmp(attr->name, "fwd_type") == 0) {
+		op = kstrtoint(buf, 10, &ft);
+
+		if(op < 0) {
+			LOG_ERR("Failed to set the reset rate.");
+			return op;
+		}
+
+		/* Single flow rate is now changed! */
+		pffb_distr = ft;
+	}
+
+	/* Renew the infos here...
+	if(pffb_distr == PFFB_NOF_DIST) {
+		memset(pffb_flows, 0, sizeof(unsigned int) * PFFB_NOP);
+	}
+	 */
+
+	return count;
+}
+
+/* Sysfs operations. */
+static const struct sysfs_ops pffb_sysfs_ops = {
+	.show  = pffb_attr_show,
+	.store = pffb_attr_store,
+};
+
+/*
+ * Attributes visible in PFFB sysfs directory.
+ */
+
+static struct pffb_attribute pffb_ft_attr =
+	__ATTR(fwd_type, 0664, pffb_attr_show, pffb_attr_store);
+
+static struct attribute * pffb_attrs[] = {
+	&pffb_ft_attr.attr,
+	NULL,
+};
+
+/* Operations associated with the release of the kobj. */
+static void pffb_release(struct kobject *kobj) {
+	/* Nothing... */
+}
+
+/* Master PFFB ktype, different for the type of shown attributes. */
+static struct kobj_type pffb_ktype = {
+	.sysfs_ops = &pffb_sysfs_ops,
+	.release = pffb_release,
+	.default_attrs = pffb_attrs,
+};
+/*
+ * Port-status related:
+ */
+
+/* Timespec to ms. */
+static inline unsigned long pffb_to_ms(struct timespec * t) {
+	return (t->tv_sec * 1000) + (t->tv_nsec / 1000000);
+}
+
+/* Just remove the entry... */
+static void __pffb_rem_flow(struct flow_info * f) {
+	list_del(&f->next);
+	rkfree(f);
+}
+
+/* Remove a flow entry with a little logic. */
+static void pffb_rem_flow(
+	struct pft_port_entry * port_e,
+	struct flow_info * flow) {
+
+	if(pffb_distr == PFFB_NOF_DIST) {
+		pffb_flows[flow->port_id] -= flow->rate;
+	}
+
+	LOG_INFO("PFFBI > %x expired on port %d, remains %d",
+		flow->key, flow->port_id, pffb_flows[flow->port_id]);
+
+	__pffb_rem_flow(flow);
+
+	return;
+}
+
+/* Find and update the flow entry. On expiration the entry is removed.
+ * Returns -1 if the flow has not found (and so is new).
+ */
+static struct pft_port_entry * pffb_find_key(
+	unsigned short key,
+	struct pft_entry * entry) {
+
+	struct pft_port_entry * p = 0;
+	struct flow_info * f = 0;
+	struct flow_info * t = 0;
+	struct timespec now;
+
+	int i = 0;
```

Comment:
got warning --> unused variable "i"
Can you remove this or  add an instruction like?
  (void)i;